File: ThreeDecks.py
t = int(input())
for _ in range(t):
    a, b, c = map(int, input().split())
    total = a + b + c
    if total % 3 != 0:
        print("NO")
    else:
        x = total // 3
        if x < a or x < b:
            print("NO")
        else:
            print("YES")

File: ThreeDecks.cpp
#include <iostream>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        long long a, b, c;
        cin >> a >> b >> c;
        
        long long total = a + b + c;
        if (total % 3 != 0) {
            cout << "NO\n";
            continue;
        }
        
        long long x = total / 3;
        
        if (x < a || x < b) {
            cout << "NO\n";
            continue;
        }
        
        cout << "YES\n";
    }
    return 0;
}
